### PR TITLE
added id generation

### PR DESCRIPTION
**Pull request by LianaHus**
removed image if empty

### CODE REVIEW

File: src/tab/tabs.ts [hunk 1]
```diff
@@ -37,28 +37,46 @@ export class RemixTabs extends LitElement {
   @property({ type: String })
   public defaultIcon: string = '';
 
+  // enables/disables + button to add new tab
   @property({ type: Boolean })
   public canAdd = true;
 
-  private defaultTab() {
+  private generateId(): string {
```
Comment (GrandSchtroumpf) [inline src/tab/tabs.ts]: I think the generation of Tab should be outside of the component. Like that we keep the component View only, and it's up to the host application to create the tab.

File: src/tab/tabs.ts [hunk 2]
```diff
@@ -37,28 +37,46 @@ export class RemixTabs extends LitElement {
...
   }
 
-  /** Add a tab to the list */
-  public addTab() {
-    const tab = this.defaultTab();
+  /**
+   * Add a tab to the list.
+   * @returns an id of the created tab.
+   * */
+  public addTab(tab: Tab): string {
```
Comment (GrandSchtroumpf) [inline src/tab/tabs.ts]: Send an event to the parent for it to handle that logic.

File: src/tab/tabs.ts [hunk 3]
```diff
@@ -37,28 +37,46 @@ export class RemixTabs extends LitElement {
...
+    }
     this.tabs = [ ...this.tabs, tab ];
+    return tab.id;
   }
 
   /** Remove a specific tab from the list */
   public removeTab({ detail: id }: CustomEvent) {
-    const index = this.tabs.indexOf(id);
-    this.tabs = [...this.tabs.slice(0, index), ...this.tabs.slice(index, this.tabs.length - 1)];
+    const index = this.tabs.indexOf(this.tabs.find(tab => tab.id === id));
```
Comment (GrandSchtroumpf) [inline src/tab/tabs.ts]: Use `findIndex` instead : 
```javascript
const index = this.tabs.findIndex(tab => tab.id === id)
```

**Comment by LianaHus** [comment]
@GrandSchtroumpf  agreed. We will move it outside as soon as we are done with testing.

**Pull request merged**